### PR TITLE
Normal distribution votes generation

### PR DESCRIPTION
**Pull request by kasptom**

### CODE REVIEW

File: brute/votemaker/tests/population.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import unittest
+from votemaker.population import Population
+
+
+class PopulationTest(unittest.TestCase):
+    def test_population(self):
+        population_01 = Population(10, 3)
+        self.assertTrue(len(population_01.voters_coordinates) == 10)
+        self.assertTrue(len(population_01.candidates_coordinates) == 3)
+if __name__ == '__main__':
```
Comment (jakubste) [inline brute/votemaker/tests/population.py]: we don't need these ifs, please delete this

File: brute/votemaker/preferences.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    def compute_preferences(self):
+        """
+            For each voter arranges candidates in order of euclidean norm distances
+            :rtype: dictionary, key: tuple of double, value: list of tuples of double
+        """
+        voters_preferences = {}
+        for voter in self.voters_coordinates:
+            voter_preferences = sorted(self.candidates_coordinates,
+                                       key=lambda candidate: self.compute_euclidean_norm(voter, candidate))
```
Comment (jakubste) [inline brute/votemaker/preferences.py]: try making sorter lines ;)

File: brute/votemaker/tests/preferences.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    def test_euclidean_norm_2(self):
+        self.assertAlmostEqual(sqrt(2), Preferences.compute_euclidean_norm((0, 0), (-1, -1)), 4)
+
+    def test_compute_preferences_1(self):
+        population_01 = Population(12, 3)
+        population_01.voters_coordinates = VOTERS_COORDINATES_01
+        population_01.candidates_coordinates = CANDIDATES_COORDINATES_01
+        preferences_01 = Preferences(population_01)
+        preferences_01.voters_preferences = preferences_01.compute_preferences()
+        self.assertListEqual(VOTERS_COORDINATES_01, population_01.voters_coordinates, "voters_coordinates - OK")
```
Comment (jakubste) [inline brute/votemaker/tests/preferences.py]: The third argument is **error** message and in majority of cases you want it to be default one.
Try yourself sth like assertListEqual([1,2,3], [1,4,3]) and see how it looks

File: brute/votemaker/tests/test_data.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+VOTERS_NUMBER_01 = 12
```
Comment (jakubste) [inline brute/votemaker/tests/test_data.py]: You don't use this, do you?

File: brute/votemaker/tests/population.py (hunk 1 above)
Comment (kasptom) [inline brute/votemaker/tests/population.py]: deleted

File: brute/votemaker/preferences.py (hunk 2 above)
Comment (kasptom) [inline brute/votemaker/preferences.py]: done

**Comment by jakubste** [comment]
Hm, just to be clear: is there any possibility to use this as data for election? Should it be separate task?

File: brute/ecs/votemaker/point.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class Point:
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    @staticmethod
+    def compute_euclidean_norm(a, b):
```
Comment (jakubste) [inline brute/ecs/votemaker/point.py]: zrób też taką na punkcie, żeby można było zrobić coś w stylu
```
a = Point(1, 2)
b = Point(3,4)
dist = a.distance(b)
```
btw calculate_distance byłoby krótsze, a wiadomo że euklidesem liczymy ;)

File: brute/ecs/votemaker/tests/test_data.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        candidates.append(candidate)
+        candidate_id += 1
+    return candidates
+
+VOTERS_01 = generate_voters()
+CANDIDATES_01 = generate_candidates()
+
+
+
+
```
Comment (jakubste) [inline brute/ecs/votemaker/tests/test_data.py]: reformat często pls

**Pull request merged**